### PR TITLE
Odyssey: Fix moment locale for en-us

### PR DESCRIPTION
**Pull request by kangzj**
Slack: p1726784616274179-slack-C04UE0ANHDY

## Proposed Changes

* Only replace php date format for simple sites

## Why are these changes being made?

* Fix date format for self hosted sites

## Testing Instructions

* Build Odyssey Stats locally
* Open `/wp-admin/admin.php?page=stats#!/stats/day/jasper1.au.ngrok.io?page=stats&flags=stats%2Fdate-picker-calendar`
* Open date range picker
* Ensure the inputs look good

|Before:|After:|
|-------|------|
|image|image|

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: apps/odyssey-stats/src/app.tsx [hunk 1]
```diff
@@ -46,6 +47,8 @@ async function AppBoot() {
 		},
 	};
 
+	const isSmiple = isSimpleSite( initialState, siteId );
```
Comment (annacmc) [inline apps/odyssey-stats/src/app.tsx]: Looks like a typo in this const

File: apps/odyssey-stats/src/app.tsx [hunk 2]
```diff
@@ -70,7 +73,7 @@ async function AppBoot() {
 	}
 
 	// Ensure locale files are loaded before rendering.
-	setLocale( localeSlug ).then( () => {
+	setLocale( localeSlug, isSmiple ).then( () => {
```
Comment (annacmc) [inline apps/odyssey-stats/src/app.tsx]: again here with the const typo

**Review by annacmc — CHANGES_REQUESTED** [review]
It looks great, nice fix. Just noted 2x typos in the constant naming that need to be fixed up.

File: apps/odyssey-stats/src/app.tsx (hunk 1 above)
Comment (kangzj) [inline apps/odyssey-stats/src/app.tsx]:
```suggestion
	const isSimple = isSimpleSite( initialState, siteId );
```

File: apps/odyssey-stats/src/app.tsx (hunk 2 above)
Comment (kangzj) [inline apps/odyssey-stats/src/app.tsx]:
```suggestion
	setLocale( localeSlug, isSimple ).then( () => {
```

**Review by annacmc — APPROVED** [review]

**Pull request merged**